modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/MvPartitionStorages.java

```diff
@@ -0,0 +1,495 @@
+            StorageOperation storageOperation = operationByPartitionId.get(partitionId);
+
+            if (storageOperation instanceof DestroyStorageOperation) {
+                destroyFutures.add(((DestroyStorageOperation) storageOperation).getDestroyFuture());
+            } else {
+                T storage = storageByPartitionId.getAndSet(partitionId, null);
+
+                if (storage != null) {
+                    destroyFutures.add(destroyStorageFunction.apply(storage));
+                }
+            }
+        }
+
+        return CompletableFuture.allOf(destroyFutures.toArray(CompletableFuture[]::new));
+    }
+
+    /**
+     * Returns table name.
+     */
+    public String getTableName() {
+        return tableView.name();
+    }
+
+    /**
+     * Creates a short info of the multi-versioned partition storage in the format "table=user, partitionId=1".
+     *
+     * @param partitionId Partition ID.
+     */
+    public String createStorageInfo(int partitionId) {
+        return IgniteStringFormatter.format("table={}, partitionId={}", getTableName(), partitionId);
+    }
+
+    /**
+     * Checks that the partition ID is within the scope of the configuration.
+     *
+     * @param partitionId Partition ID.
+     * @throws IllegalArgumentException If partition ID is out of configured bounds.
+     */
+    private void checkPartitionId(int partitionId) {
+        int partitions = storageByPartitionId.length();
+
+        if (partitionId < 0 || partitionId >= partitions) {
+            throw new IllegalArgumentException(IgniteStringFormatter.format(
+                    "Unable to access partition with id outside of configured range: [table={}, partitionId={}, partitions={}]",
+                    getTableName(),
+                    partitionId,
+                    partitions
+            ));
+        }
+    }
+
+    /**
+     * Checks if the storage exists.
+     *
+     * @param partitionId Partition ID.
+     * @throws IllegalArgumentException If partition ID is out of configured bounds.
+     * @throws StorageException If the storage does not exist.
+     */
+    private void checkStorageExists(int partitionId) {
+        if (get(partitionId) == null) {
+            throw new StorageException(createStorageDoesNotExistErrorMessage(partitionId));
+        }
+    }
+
+    /**
+     * Checks if the storage exists.
+     *
+     * @param partitionId Partition ID.
+     * @throws IllegalArgumentException If partition ID is out of configured bounds.
+     * @throws StorageRebalanceException If the storage does not exist.
+     */
+    private void checkStorageExistsForRebalance(int partitionId) {
+        if (get(partitionId) == null) {
+            throw new StorageRebalanceException(createStorageDoesNotExistErrorMessage(partitionId));
+        }
+    }
+
+    private void throwExceptionDependingOnOperation(StorageOperation operation, int partitionId) {
+        if (operation == null) {
```

Comment:
Maybe we can extract null check, otherwise it looks like this method always throws which is not the case

Reply:
I agree, I'll fix it.